From bd2edfd03f463c3167fbd084b2d8bde6ad0eb885 Mon Sep 17 00:00:00 2001
From: Li Lingfeng <lilingfeng3@huawei.com>
Date: Tue, 10 May 2022 23:05:10 +0800
Subject: [PATCH] block: fix kabi change since add bd_write_openers and
 bd_part_write_openers
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: bd2edfd03f463c3167fbd084b2d8bde6ad0eb885
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I53Q6M
CVE: NA

---------------------------

Signed-off-by: Li Lingfeng <lilingfeng3@huawei.com>
Reviewed-by: Zhang Yi <yi.zhang@huawei.com>
Reviewed-by: zhihao Cheng <chengzhihao1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/blk_types.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/linux/blk_types.h b/include/linux/blk_types.h
index 5410050d5017..bbb62ff84601 100644
--- a/include/linux/blk_types.h
+++ b/include/linux/blk_types.h
@@ -48,9 +48,11 @@ struct block_device {
 	int			bd_fsfreeze_count;
 	/* Mutex for freeze */
 	struct mutex		bd_fsfreeze_mutex;
-	int			bd_write_openers;
-	int			bd_part_write_openers;
+#ifndef __GENKSYMS__
+	KABI_USE2(1, int bd_write_openers, int bd_part_write_openers);
+#else
 	KABI_RESERVE(1)
+#endif
 	KABI_RESERVE(2)
 	KABI_RESERVE(3)
 	KABI_RESERVE(4)
-- 
2.33.0

